common/include/pcl/point_cloud.h

```diff
@@ -491,7 +491,7 @@ namespace pcl
       push_back (const PointT& pt)
       {
         points.push_back (pt);
-        width = static_cast<std::uint32_t> (points.size ());
+        width = static_cast<index_t> (points.size ());
```

Comment:
Replace with call to `size()` instead with a note to remove these casts later on